### PR TITLE
Vsphere module: collect custom fields from virtual machines (#4464)

### PR DESCRIPTION
**Pull request by amandahla**
To obtain custom fields values. 
The test is missing this part because simulator doesn't have custom field manager.

### CODE REVIEW

**Comment by amandahla** [comment]
I think I made a little mess using rebase...My commit to fix used memory field is here too. If it's a problem, I can try to remove him. :-|

**Comment by exekias** [comment]
jenkins test it please

**Comment by exekias** [comment]
@amandahla you can always try to remove it using `git rebase -i HEAD~3`, but you have to be careful, anyway it doesn't matter if the other one goes in

File: metricbeat/module/vsphere/virtualmachine/virtualmachine.go [hunk 1]
```diff
@@ -66,6 +68,27 @@ func (m *MetricSet) Fetch() ([]common.MapStr, error) {
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
+	// Get all custom fields.
+	customFieldsOk := true
+
+	fieldMap := make(map[int32]string)
```
Comment (exekias) [inline metricbeat/module/vsphere/virtualmachine/virtualmachine.go]: This name can be misleading, perhaps `customFieldsMap` is more clear? what do you think?

**Comment by amandahla** [comment]
@exekias Good point. I didn't think about it. I'll make the changes (including the name of the field to customFieldsMap). 
I didn't found in the docs if there is a limit so it would be nice/safer to leave this optional.

**Comment by exekias** [comment]
LGTM, jenkins test it

**Comment by exekias** [comment]
@amandahla it seems test are failing because of an issue fixed in master, could you please rebase master so it can pass tests?

**Comment by amandahla** [comment]
@exekias Is something that I'm missing here? :-(
I compare the two and didn't found differences 
https://github.com/elastic/beats/compare/master...estaleiro:iss4464

**Comment by exekias** [comment]
jenkins test this please

**Comment by monicasarbu** [comment]
@amandahla Can you please rebase to master?

**Comment by amandahla** [comment]
There is a error on building filebeat :| I'll do another rebase.

**Comment by exekias** [comment]
Hi @amandahla, now that https://github.com/elastic/beats/pull/4462 is finally merged, perhaps you can rebase this PR? I think the error will go away

**Pull request merged**